src/main/scala/cromwell/Main.scala

```diff
@@ -21,167 +20,222 @@ object Actions extends Enumeration {
+      val inputsPath = argPath(args, 1, Option(".inputs"), checkDefaultExists = false)
+      val optionsPath = argPath(args, 2, Option(".options"), checkDefaultExists = true)
+      val metadataPath = argPath(args, 3, None)
 
-      val inputs: binding.WorkflowRawInputs = jsValue match {
-        case JsObject(rawInputs) => rawInputs
-        case _ => throw new RuntimeException("Expecting a JSON object")
-      }
+      Log.info(s"Default backend: ${WorkflowManagerActor.BackendType}")
+      Log.info(s"RUN sub-command")
+      Log.info(s"  $WdlLabel: $wdlPath")
+      inputsPath.foreach(path => Log.info(s"  $InputsLabel: $path"))
+      optionsPath.foreach(path => Log.info(s"  $OptionsLabel: $path"))
+      metadataPath.foreach(path => Log.info(s"  $MetadataLabel: $path"))
 
-      val options = WorkflowOptions.fromJsonObject(workflowOptions.parseJson.asInstanceOf[JsObject]) match {
-        case Success(x) => x
+      checkPaths(wdlPath, inputsPath, optionsPath, metadataPath) match {
+        case Success(workflowSourceFiles) => runWorkflow(workflowSourceFiles, metadataPath)
         case Failure(ex) =>
-          System.err.println(s"ERROR: Could not parse workflow options:\n")
-          System.err.println(ex.getMessage)
-          System.exit(1)
-          ???
+          Console.err.println(ex.getMessage)
+          exit(1)
       }
+    }
+  }
 
-      inputs foreach { case (k, v) => Log.info(s"input: $k => $v") }
-      val sources = WorkflowSourceFiles(wdlSource, inputsJson, options.asPrettyJson)
-      val singleWorkflowRunner = SingleWorkflowRunnerActor.props(sources, inputs, workflowManagerSystem.workflowManagerActor)
-      workflowManagerSystem.actorSystem.actorOf(singleWorkflowRunner, "SingleWorkflowRunnerActor")
-      workflowManagerSystem.actorSystem.awaitTermination()
-      // And now we just wait for the magic to happen
-    } catch {
-      case e: Exception =>
-        println("Unable to process inputs")
-        e.printStackTrace()
+  /**
+    * Retrieve the arg at index as path, or return some default. Args specified as "-" will be returned as None.
+    *
+    * @param args The run command arguments, with the wdl path at arg.head.
+    * @param index The index of the path we're looking for.
+    * @param defaultExt The default extension to use if the argument was not specified at all.
+    * @param checkDefaultExists If true, verify that our computed default file exists before using it.
+    * @return The argument as a Path resolved as a sibling to the wdl path.
+    */
+  private[this] def argPath(args: Seq[String], index: Int, defaultExt: Option[String],
+                            checkDefaultExists: Boolean = true): Option[Path] = {
+
+    // To return a default, swap the extension, and then maybe check if the file exists.
+    def defaultPath = defaultExt
+      .map(ext => swapExt(args.head, ".wdl", ext))
+      .filter(path => !checkDefaultExists || Files.exists(Paths.get(path)))
```

Comment:
I like to use `canRead` instead of `exists` in those cases where you actually want to access the file.